src/test/java/net/bhardy/bizzo/billing/PolicyBuilderTest.java

```diff
@@ -60,6 +137,93 @@ public void isDueOn_monthlyFiltered() {
     }
 
     @Test
+    public void upcomingDueDates_dailySimple() {
+        BillingPolicy policy = BillingPolicy.builder()
+                .everyDay()
+                .build();
+
+        LocalDate oneSaturday = LocalDate.of(2017, 8, 5);
+        List<LocalDate> nextBills = policy.upcomingDueDates(oneSaturday, 2);
+
+        LocalDate expected1 = LocalDate.of(2017, 8, 5);
+        LocalDate actual1 = nextBills.get(0);
+        assertEquals(expected1, actual1);
+        assertEquals(SATURDAY, actual1.getDayOfWeek());
+
+        LocalDate expected2 = LocalDate.of(2017, 8, 6);
+        LocalDate actual2 = nextBills.get(1);
+        assertEquals(expected2, actual2);
+        assertEquals(SUNDAY, actual2.getDayOfWeek());
+    }
+
+    @Ignore  // this reveals an underlying bug we need to fix
+    @Test
+    public void upcomingDueDates_dailyFiltered() {
+        BillingPolicy policy = BillingPolicy.builder()
+                .everyDay()
+                .filter(not(daysOfWeek(SATURDAY, SUNDAY)))
+                .action(ActionChoice.Kind.NEXT_DAY)
+                .build();
+
+        LocalDate oneSaturday = LocalDate.of(2017, 8, 5);
+
+        List<LocalDate> nextBills = policy.upcomingDueDates(oneSaturday, 2);
+
+        LocalDate expected1 = LocalDate.of(2017, 8, 7);
+        LocalDate actual1 = nextBills.get(0);
+        assertEquals(expected1, actual1);
+        assertEquals(MONDAY, actual1.getDayOfWeek());
+
+        LocalDate expected2 = LocalDate.of(2017, 8, 8);
+        LocalDate actual2 = nextBills.get(1);
+        assertEquals(expected2, actual2);
+        assertEquals(TUESDAY, actual2.getDayOfWeek());
+    }
+
+    @Test
+    public void upcomingDueDates_weeklySimple() {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)